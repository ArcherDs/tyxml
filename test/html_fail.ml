open Tyxml.Html ;;

div [a [a []]] ;;

div [address [a [address []]]];;
