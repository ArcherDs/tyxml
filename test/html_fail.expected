Characters 6-14:
  div [a [a []]] ;;
       ^^^^^^^^
Error: This expression has type
         ([> `A of [> `A of 'c ] as 'b ] as 'a) Html5.elt
       but an expression was expected of type
         ([< Html5_types.div_content_fun ] as 'd) Html5.elt
       Type 'a is not compatible with type
         'd =
           [< `A of Html5_types.flow5_without_interactive
            | `Abbr
            | `Address
            | `Article
            | `Aside
            | `Audio of Html5_types.flow5_without_media
            | `Audio_interactive of Html5_types.flow5_without_media
            | `B
            | `Bdo
            | `Blockquote
            | `Br
            | `Button
            | `Canvas of Html5_types.flow5
            | `Cite
            | `Code
            | `Command
            | `Datalist
            | `Del of Html5_types.flow5
            | `Details
            | `Dfn
            | `Div
            | `Dl
            | `Em
            | `Embed
            | `Fieldset
            | `Figure
            | `Footer
            | `Form
            | `H1
            | `H2
            | `H3
            | `H4
            | `H5
            | `H6
            | `Header
            | `Hgroup
            | `Hr
            | `I
            | `Iframe
            | `Img
            | `Img_interactive
            | `Input
            | `Ins of Html5_types.flow5
            | `Kbd
            | `Keygen
            | `Label
            | `Main
            | `Map of Html5_types.flow5
            | `Mark
            | `Menu
            | `Meter
            | `Nav
            | `Noscript of Html5_types.flow5_without_noscript
            | `Object of Html5_types.flow5
            | `Object_interactive of Html5_types.flow5
            | `Ol
            | `Output
            | `P
            | `PCDATA
            | `Pre
            | `Progress
            | `Q
            | `Ruby
            | `Samp
            | `Script
            | `Section
            | `Select
            | `Small
            | `Span
            | `Strong
            | `Style
            | `Sub
            | `Sup
            | `Svg
            | `Table
            | `Textarea
            | `Time
            | `U
            | `Ul
            | `Var
            | `Video of Html5_types.flow5_without_media
            | `Video_interactive of Html5_types.flow5_without_media
            | `Wbr ] 
       Type 'b is not compatible with type
         Html5_types.flow5_without_interactive =
           [ `Abbr
           | `Address
           | `Article
           | `Aside
           | `Audio of Html5_types.flow5_without_media
           | `B
           | `Bdo
           | `Blockquote
           | `Br
           | `Button
           | `Canvas of Html5_types.flow5
           | `Cite
           | `Code
           | `Command
           | `Datalist
           | `Del of Html5_types.flow5
           | `Dfn
           | `Div
           | `Dl
           | `Em
           | `Fieldset
           | `Figure
           | `Footer
           | `Form
           | `H1
           | `H2
           | `H3
           | `H4
           | `H5
           | `H6
           | `Header
           | `Hgroup
           | `Hr
           | `I
           | `Img
           | `Input
           | `Ins of Html5_types.flow5
           | `Kbd
           | `Keygen
           | `Label
           | `Main
           | `Map of Html5_types.flow5
           | `Mark
           | `Menu
           | `Meter
           | `Nav
           | `Noscript of Html5_types.flow5_without_noscript
           | `Object of Html5_types.flow5
           | `Ol
           | `Output
           | `P
           | `PCDATA
           | `Pre
           | `Progress
           | `Q
           | `Ruby
           | `Samp
           | `Script
           | `Section
           | `Select
           | `Small
           | `Span
           | `Strong
           | `Style
           | `Sub
           | `Sup
           | `Svg
           | `Table
           | `Textarea
           | `Time
           | `U
           | `Ul
           | `Var
           | `Video of Html5_types.flow5_without_media
           | `Wbr ] 
       The second variant type does not allow tag(s) `A

